File: tests/org/elixir_lang/reference/callable/Issue456Test.java
package org.elixir_lang.reference.callable;

import com.intellij.psi.PsiElement;
import org.elixir_lang.PlatformTestCase;
import org.elixir_lang.psi.call.Call;

import static org.elixir_lang.reference.Callable.isVariable;

public class Issue456Test extends PlatformTestCase {
    /*
     * Tests
     */

    public void testIsVariable() {
        myFixture.configureByFiles("is_variable.ex");
        PsiElement callable = myFixture
                .getFile()
                .findElementAt(myFixture.getCaretOffset())
                .getPrevSibling();

        assertInstanceOf(callable, Call.class);
        assertFalse("order_by is incorrectly marked as a variable", isVariable(callable));
    }

    /*
     * Protected Instance Methods
     */

    @Override
    protected String getTestDataPath() {
        return "testData/org/elixir_lang/reference/callable/issue_456";
    }
}

File: tests/org/elixir_lang/reference/callable/Issue456Test.kt
package org.elixir_lang.reference.callable

import org.elixir_lang.PlatformTestCase
import org.elixir_lang.psi.call.Call
import org.elixir_lang.reference.Callable.Companion.isVariable

class Issue456Test : PlatformTestCase() {
    fun testIsVariable() {
        myFixture.configureByFiles("is_variable.ex")
        val callable = myFixture
                .file
                .findElementAt(myFixture.caretOffset)!!
                .prevSibling
        assertInstanceOf(callable, Call::class.java)
        assertFalse("order_by is incorrectly marked as a variable", isVariable(callable))
    }

    override fun getTestDataPath(): String = "testData/org/elixir_lang/reference/callable/issue_456"
}
